Resetcontrol

peteasa opened the pull request:
During reset several control signals are not cleared.  This merge fixes this, tidies up the mailbox and adds a missing include.

Signed-off-by: Peter Saunderson peteasa@gmail.com

peteasa committed:
Add missing include
Signed-off-by: Peter Saunderson <peteasa@gmail.com>

peteasa committed:
Tidy up mailbox

Signed-off-by: Peter Saunderson <peteasa@gmail.com>

peteasa committed:
Ensure clock during reset to fix etx_fifo.txrd_fifo reset

Signed-off-by: Peter Saunderson <peteasa@gmail.com>

peteasa committed:
Applied reset to several control registers

Signed-off-by: Peter Saunderson <peteasa@gmail.com>

aolofsson commented:
Sorry for being away for a while...I don't feel comfortable with some of these changes. In particular, the etx_clocks is a red flag. The following statement probably won't synthesize correctlyas it's mixing async and sync reset. Not sure what the tool will do here? Might work in sim but won't work on fpga.  

"if(!sys_nreset & ~(heartbeat | heartbeat_double))"

Feels like the root cause is still open,looking at it now.

peteasa commented:
I removed the reset changes from my other pull requests to avoid confusion.  Also wanted to see if the test run with the fpga had any differences without the clock changes.. no change detected...

In the simulation runs these reset changes are good because it enables the new test harness to write to the mailbox.... without them too much is left undefined to work out what is going on.  I will do something local for the time being to keep the simulation working and wait for a better fix..

I moved the mailbox tidy up into the latest pull request.. thus it seemed appropriate to close this pull request.
